apps/website/app/utils/llm/cors.ts

```diff
@@ -0,0 +1,48 @@
+import { NextRequest } from "next/server";
+
+const allowedOrigins = ["https://roamresearch.com", "http://localhost:3000"];
+
+function isVercelPreviewUrl(origin: string | null): boolean {
+  if (!origin) return false;
+  return origin.includes(".vercel.app") || origin.includes("discourse-graph");
+}
+
+function isAllowedOrigin(origin: string | null): boolean {
+  if (!origin) return false;
+  return (
+    allowedOrigins.some((allowed) => origin.startsWith(allowed)) ||
+    isVercelPreviewUrl(origin)
+  );
+}
+
+export default function cors(req: NextRequest, res: Response) {
+  const origin = req.headers.get("origin");
+
+  const originIsAllowed = isAllowedOrigin(origin);
```

Comment:
remove origin from the function params and change this to `origin && isAllowedOrigin(origin)`